specification/src/test/groovy/io/datanerds/rappelkiste/specification/PatchCounterSpecification.groovy

```diff
@@ -0,0 +1,310 @@
+        URI postUri = new URI(baseUrl + counterPath)
+        def JsonNode uuid = objectMapper.readTree(RestAssured.post(postUri).body.asString())
+
+        def patchTo = [
+                "op"   : "add",
+                "path" : "path",
+                "value": "-1"
+        ]
+
+        when: "A counter is being Patched"
+        URI patchUri = new URI(baseUrl + counterPath + "/" + uuid.asText())
+        def Response response = RestAssured
+                .given()
+                .contentType("application/json-patch+json")
+                .body(patchTo)
+                .patch(patchUri)
+
+        //todo @fw should this be 204??
+        then: "The Patch request has status code 200"
+        assertThat(response.statusCode, is(200))
+
+        and: "The counter has been updated by one"
+        await().ignoreExceptions().until({
+            RestAssured.given().get(patchUri).then().assertThat().body(equalTo("-1"))
+        })
+
+        where:
+        baseUrl << servers
+
+    }
+
+    def "Patching an existing Counter by adding -1 and 1"(String baseUrl) {
+
+        given: "A preexisting counter, with a uuid and a PatchOperation Object"
+        URI postUri = new URI(baseUrl + counterPath)
+        def JsonNode uuid = objectMapper.readTree(RestAssured.post(postUri).body.asString())
+
+        def patchTo1 = [
+                "op"   : "add",
+                "path" : "path",
+                "value": "1"
+        ]
+
+        def patchTo2 = [
+                "op"   : "add",
+                "path" : "path",
+                "value": "-1"
+        ]
+
+        when: "A counter is being Patched"
+        URI patchUri = new URI(baseUrl + counterPath + "/" + uuid.asText())
+        def Response response1 = RestAssured
+                .given()
+                .contentType("application/json-patch+json")
+                .body(patchTo1)
+                .patch(patchUri)
+
+        def Response response2 = RestAssured
+                .given()
+                .contentType("application/json-patch+json")
+                .body(patchTo2)
+                .patch(patchUri)
+
+        //todo @fw should this be 204??
+        then: "The first Patch request has status code 200"
+        assertThat(response1.statusCode, is(200))
+
+        //todo @fw should this be 204??
+        and: "The second Patch request has status code 200"
+        assertThat(response2.statusCode, is(200))
+
+        and: "The counter has been updated by one"
+        await().ignoreExceptions().until({
+            RestAssured.given().get(patchUri).then().assertThat().body(equalTo("0"))
+        })
+
+        where:
+        baseUrl << servers
+
+    }
+
+
+    def "Patching an existing Counter by setting it to 42"(String baseUrl) {
+
+        given: "A preexisting counter, with a uuid and a PatchOperation Object"
+        URI postUri = new URI(baseUrl + counterPath)
+        def JsonNode uuid = objectMapper.readTree(RestAssured.post(postUri).body.asString())
```

Comment:
Probably a trait makes sense since you're using this pretty often
```
def uuid = RestAssured
                    .expect()
                        .statusCode(201)
                    .when()
                        .post(baseUrl + counterPath)
                    .andReturn()
                        .as(UUID.class)
```